src/event.rs

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//! This is an abstract EventNotifer and EventConsumer for eventfd and pipefd
+
+use std::io::{Read, Write};
+use std::{
+    fs::File,
+    io,
+    os::fd::{AsRawFd, FromRawFd},
+    result,
+};
+
+/// EventNotifier
+/// This is a generic event notifier that can be used with eventfd or pipefd.
+/// It allows writing a value to the file descriptor to notify an event.
+#[derive(Debug)]
+pub struct EventNotifier {
+    fd: File,
+}
+
+impl EventNotifier {
+    /// Write a value to the EventNotifier's fd
```

Comment:
This is not writing a generic "value", but a specific `1`, I guess to be compatible with eventfd, right?
I think we should document that.